File: stimuli.js
function shuffle(array) {
  var currentIndex = array.length, randomIndex;

  while (currentIndex != 0) {
    randomIndex = Math.floor(Math.random() * currentIndex);
    currentIndex--;

    [array[currentIndex], array[randomIndex]] = [array[randomIndex], array[currentIndex]];
  }

  return array;
}

const nouns = [
  { noun: "pallo", eng: "ball", freq: "high_frequency", stem: "no_stem_change" },
  { noun: "avain", eng: "key", freq: "high_frequency", stem: "stem_change" },
  { noun: "hammasharja", eng: "toothbrush", freq: "low_frequency", stem: "no_stem_change" },
  { noun: "jakoavain", eng: "wrench", freq: "low_frequency", stem: "stem_change" }
];

const embedded = ['lose', 'forget'];

const embeddingAct = ['thought', 'said', 'knew'];

const embeddingPass = ['was.thought', 'was.said', 'was.known'];

const preps = ['in.village', 'in.town', 'in.neighborhood'];

const nomNames = ["Pekka", "Arto", "Helvi", "Antti", "Otto", "Kari"];

const genNames = ["Eeron", "Matin", "Mikon", "Tuulin", "Jorman", "Jaanan"];

let shuffledNouns = [...nouns];
shuffle(shuffledNouns);

// Add console.log statements for debugging or verification
console.log('Shuffled Nouns:', shuffledNouns);

const nounsActive = shuffledNouns.slice(0, 2);
const nounsPassive = shuffledNouns.slice(2);

console.log('Nouns Active:', nounsActive);
console.log('Nouns Passive:', nounsPassive);

function makeActiveStims(listDicts) {
  const activeStims = [];
  for (const dict of listDicts) {
    let stim = '_';
    const embeddedV = [embedded[Math.floor(Math.random() * embedded.length)]];
    stim = embeddedV[0] + ' ' + stim;
    const genName = [genNames[Math.floor(Math.random() * genNames.length)]];
    stim = genName[0] + ' ' + stim;
    const embedding = [embeddingAct[Math.floor(Math.random() * embeddingAct.length)]];
    stim = embedding[0] + ' ' + stim;
    const nomName = [nomNames[Math.floor(Math.random() * nomNames.length)]];
    stim = nomName[0] + ' ' + stim;
    const dictNew = { stim };
    const dictFinal = { ...dictNew, ...dict };
    activeStims.push(dictFinal);
  }
  return activeStims;
}

// Add console.log statements after function calls
console.log('Active Stims:', makeActiveStims(nounsActive));

function makePassiveStims(listDicts) {
  const passiveStims = [];
  for (const dict of listDicts) {
    let stim = '_';
    const embeddedV = [embedded[Math.floor(Math.random() * embedded.length)]];
    stim = embeddedV[0] + ' ' + stim;
    const genName = [genNames[Math.floor(Math.random() * genNames.length)]];
    stim = genName[0] + ' ' + stim;
    const embedding = [embeddingPass[Math.floor(Math.random() * embeddingPass.length)]];
    stim = embedding[0] + ' ' + stim;
    const prep = [preps[Math.floor(Math.random() * preps.length)]];
    stim = prep[0] + ' ' + stim;
    const dictNew = { stim };
    const dictFinal = { ...dictNew, ...dict };
    passiveStims.push(dictFinal);
  }
  return passiveStims;
}

// Add console.log statements after function calls
console.log('Passive Stims:', makePassiveStims(nounsPassive));

function makeAllStims(list1, list2) {
  const allStims = makeActiveStims(list1).concat(makePassiveStims(list2));
  return allStims;
}

// Add console.log statements after function calls
console.log('All Stims:', makeAllStims(nounsActive, nounsPassive));

  

File: stimuli.py
import random
import json

nouns = [{'noun': "pallo", 'eng': "ball", 'freq': "high_frequency", 'stem': "no_stem_change"}, 
{'noun': "avain", 'eng': "key", 'freq': "high_frequency", 'stem': "stem_change"}, 
{'noun': "hammasharja", 'eng': "toothbrush", 'freq': "low_frequency", 'stem': "no_stem_change"},
{'noun': "jakoavain", 'eng': "wrench", 'freq': "low_frequency", 'stem': "stem_change"},]

embedded = ['lose', 'forget']

embedding_act = ['thought', 'said', 'knew']

embedding_pass = ['was.thought', 'was.said', 'was.known']

preps = ['in.village', 'in.town', 'in.neighborhood']

nom_names = ["Pekka", "Arto", "Helvi", "Antti", "Otto", "Kari"]

gen_names = ["Eeron", "Matin", "Mikon", "Tuulin", "Jorman", "Jaanan"]

shuffled_nouns = nouns[:]
random.shuffle(shuffled_nouns)

nouns_active = shuffled_nouns[:2]
nouns_passive = shuffled_nouns[2:]


def make_active_stims(list_dicts):
    active_stims = []
    for dict in list_dicts:
        stim = ''
        stim += '_'
        embedded_v = random.sample(embedded, 1)
        stim = embedded_v[0] + " " + stim
        gen_name = random.sample(gen_names, 1)
        stim = gen_name[0] + " " + stim
        embedding = random.sample(embedding_act, 1)
        stim = embedding[0] + " " + stim
        nom_name = random.sample(nom_names, 1)
        stim = nom_name[0] + " " + stim
        dict_new = {'stim': stim} 
        dict_final = {**dict_new, **dict}
        active_stims.append(dict_final)
    return active_stims

make_active_stims(nouns_active)

def make_passive_stims(list_dicts):
    passive_stims = []
    for dict in list_dicts:
        stim = ''
        stim += '_'
        embedded_v = random.sample(embedded, 1)
        stim = embedded_v[0] + " " + stim
        gen_name = random.sample(gen_names, 1)
        stim = gen_name[0] + " " + stim
        embedding = random.sample(embedding_pass, 1)
        stim = embedding[0] + " " + stim
        prep = random.sample(preps, 1)
        stim = prep[0] + " " + stim
        dict_new = {'stim': stim}
        dict_final = {**dict_new, **dict}
        passive_stims.append(dict_final)
    return passive_stims

make_passive_stims(nouns_passive)

def make_all_stims(list1, list2):
    all_stims = make_active_stims(list1) + make_passive_stims(list2)
    return(all_stims)

make_all_stims(nouns_active, nouns_passive)


import random 

names = ["John", "Mary"]
verbs = ["runs,", "walks"]

def create_stimuli(list_names, list_verbs):
    list_stims = []
    for name in list_names:
        stim = ""
        verb = random.sample(list_verbs, 1)[0]
        print(name)
        print(verb)
        stim = name + " " + verb
        list_stims.append(stim)
    return list_stims

create_stimuli(names, verbs)
        






